### PR TITLE
merge from upstream --with-extra-version config option and set default

### PR DESCRIPTION
**Pull request by larham**
- from postgres 9.5, cherry-pick commit that adds --with-extra-version config option 
- set default to "-oss" for appending to Greenplum version

### CODE REVIEW

**Comment by schubert** [comment]
Can we change this to just match upstream postgres `--with-extra-version=STRING`

**Comment by larham** [comment]
@schubert thanks, we updated both name and PR comments.

File: configure.in [hunk 1]
```diff
@@ -34,11 +34,15 @@ AC_CONFIG_AUX_DIR(config)
...
 
-AC_DEFINE_UNQUOTED(PG_VERSION, "$PG_PACKAGE_VERSION", [Postgres version Greenplum Database is based on])
 [PG_MAJORVERSION=`expr "$PG_PACKAGE_VERSION" : '\([0-9][0-9]*\.[0-9][0-9]*\)'`]
 AC_SUBST(PG_MAJORVERSION)
 AC_DEFINE_UNQUOTED(PG_MAJORVERSION, "$PG_MAJORVERSION", [PostgreSQL major version as a string])
 
+PGAC_ARG_REQ(with, extra-version, [STRING], [append STRING to version],
+             [PG_VERSION="$PG_PACKAGE_VERSION$withval"],
+             [PG_VERSION="$PG_PACKAGE_VERSION"])
+AC_DEFINE_UNQUOTED(PG_VERSION, "$PG_VERSION", [PostgreSQL version as a string])
```
Comment (d) [inline configure.in]: The help text for `PG_VERSION` should probably remain `Postgres version Greenplum Database is based on`

Comment (larham) [reply configure.in]: Nice catch, fixing...

**Comment by larham** [comment]
@d regarding where the version shows up, our acceptance criteria from @ivannovick was just reconfirmed in person:  the with_extra_version suffix needs to show up from all tools/utilities/sql.

For example, the output from `gpssh --version` is the Greenplum version *only*, e.g.:

```
gpssh version 6.0.0 alpha.0+dev.1591.g182c71f build commit:182c71fa8206b5b172b7a0e43569dc6bb32c59f9
```

In other words, is not sufficient to have just the #4004 PR which adds to the postgres version, we need to incorporate the suffix into the Greenplum version.

So I see at least two alternatives:

A) append with_extra_version to both the postgres version and the Greenplum version , such that it shows up twice from `select version()`

B) remove the addition of with_extra_version to the postgres version, and only append it to the Greenplum version. 

I favor B since the configuration is about building Greenplum, with the assumption that the underlying postgres version is relatively stable for a given major Greenplum release. However, it would change less upstream to have it show up in both places, and Ivan would still be ok with that.

**Comment by goutamtadi1** [comment]
Created a new PR #4013. Closing this.

**Pull request closed without merging**